### PR TITLE
Miceru/fix for automanage completeness

### PR DESCRIPTION
**Pull request by mc3784**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [ ] update existing version for new feature
      - [x] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific language SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [x] No refresh required for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following apply to the PR so that label "WaitForARMFeedback" will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you are using OpenAPIHub to initialize the PR for adding a new version. More details, refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by raosuhas** [comment]
This is a breaking change Please add a new api version as 

> NewApiVersionRequired reason: A service’s API is a contract with customers and is represented by using the api-version query parameter. Changes such as adding an optional property to a request/response or introducing a new operation is a change to the service’s contract and therefore requires a new api-version value. This is critically important for documentation, client libraries, and customer support. EXAMPLE: if a customer calls a service in the public cloud using api-version=2020-07-27, the new property or operation may exist but if they call the service in a government cloud, air-gapped cloud, or Azure Stack Hub cloud using the same api-version, the property or operation may not exist. Because there is no clear relationship between the service api-version and the new property/operation, customers can’t trust the documentation and Azure customer have difficulty helping customers diagnose issues. In addition, each client library version documents the service version it supports. When an optional property or new operation is added to a service and its Swagger, new client libraries must be produced to expose this functionality to customers. Without updating the api-version, it is unclear to customers which version of a client library supports these new features.

Please fix this comment. This change requires a new api version as explained above

File: specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-04-30-preview/automanage.json [hunk 1]
```diff
@@ -1370,6 +1370,602 @@
...
+        },
+        "produces": [
+          "application/json"
+        ],
+        "consumes": [
+          "application/json"
+        ]
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.AzureStackHci/clusters/{clusterName}/providers/Microsoft.Automanage/configurationProfileAssignments/{configurationProfileAssignmentName}": {
```
Comment (raosuhas) [inline specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-04-30-preview/automanage.json:1672]: >configurationProfileAssignments [](http://example.com/codeflow?start=160&length=31)

collection GET must be implemented even for this configurationProfileAssignments resource not just the reports

**Review by raosuhas — CHANGES_REQUESTED** [review]
:clock1:

File: specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-04-30-preview/automanage.json [hunk 2]
```diff
@@ -1370,6 +1370,602 @@
...
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
```
Comment (raosuhas) [inline specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-04-30-preview/automanage.json:1662]: >null [](http://example.com/codeflow?start=26&length=4)

Why is this set to null ?

File: specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-04-30-preview/automanage.json [hunk 3]
```diff
@@ -1370,6 +1370,602 @@
...
```
Comment (raosuhas) [inline specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-04-30-preview/automanage.json:1960]: >null [](http://example.com/codeflow?start=26&length=4)

same here ?

File: specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-04-30-preview/automanage.json (hunk 2 above)
Comment (mc3784) [reply specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-04-30-preview/automanage.json]: Hi, we will implement paging and we have backlog story for that. We followed instruction as described in the documentation here:
https://armwiki.azurewebsites.net/api_contracts/guidelines/openapi.html?q=x-ms-pageable

If x-ms-pageable is added after the fact it becomes a breaking change. x-ms-pageable can be added to APIs that do not yet support pagination by setting nextLinkName to null. In this way you open yourself up for making the API paginated in the future without breaking existing clients.

File: specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-04-30-preview/automanage.json (hunk 3 above)
Comment (mc3784) [reply specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-04-30-preview/automanage.json:2113]: Hi, we will implement paging and we have backlog story for that. We followed instruction as described in the documentation here:
https://armwiki.azurewebsites.net/api_contracts/guidelines/openapi.html?q=x-ms-pageable

If x-ms-pageable is added after the fact it becomes a breaking change. x-ms-pageable can be added to APIs that do not yet support pagination by setting nextLinkName to null. In this way you open yourself up for making the API paginated in the future without breaking existing clients.

File: specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-04-30-preview/automanage.json (hunk 1 above)
Comment (mc3784) [reply specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-04-30-preview/automanage.json:1774]: Thanks! I've added this in the latest push.

File: specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-04-30-preview/automanage.json (hunk 2 above)
Comment (raosuhas) [reply specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-04-30-preview/automanage.json]: Sounds good.

File: specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-04-30-preview/automanage.json (hunk 3 above)
Comment (raosuhas) [reply specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-04-30-preview/automanage.json:2113]: Sounds good. Thanks

File: specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-04-30-preview/automanage.json [hunk 4]
```diff
@@ -1037,26 +1088,614 @@
...
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK"
+          },
+          "204": {
+            "description": "OK"
```
Comment (raosuhas) [inline specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-04-30-preview/automanage.json:1585]: >OK [](http://example.com/codeflow?start=28&length=2)

NoContent

**Review by raosuhas — APPROVED** [review]
:shipit:

**Review by ArcturusZhang — APPROVED** [review]

**Review by msyyc — APPROVED** [review]

**Pull request merged**